### PR TITLE
SAIC-602 Fix the filtering by tenant id

### PR DESCRIPTION
**Pull request by SVilgelm**
Add the subnets from the shared networks.
Filter the routers by the subnets.

### CODE REVIEW

**Comment by roman-verchikov** [comment]
Looks good, let's wait for staging test results

File: cloudferrylib/os/network/neutron.py [hunk 1]
```diff
@@ -86,9 +86,26 @@ def read_info(self, **kwargs):
...
+
+        routers = []
+        subnet_ids = set([sn['id'] for sn in subnets])
+        for router in self.get_routers():
+            router_subnet_ids = subnet_ids & set(router['subnet_ids'])
+            if router_subnet_ids:
+                router['subnet_ids'] = list(router_subnet_ids)
+                LOG.debug('Add router: %s', router['id'])
+                routers.append(router)
+                continue
```
Comment (miarmak) [inline cloudferrylib/os/network/neutron.py]: Do we really need `continue` here?

**Comment by miarmak** [comment]
Looks good in general. Seems it also can fix routers problem (SAIC-577)...
@ntymtsiv could you also verify it please?

**Comment by ntymtsiv** [comment]
+1, works good

File: cloudferrylib/os/network/neutron.py (hunk 1 above)
Comment (SVilgelm) [inline cloudferrylib/os/network/neutron.py]: No, this is a piece of the old code, has to be removed. Thank you.

Comment (SVilgelm) [inline cloudferrylib/os/network/neutron.py]: Done

**Comment by roman-verchikov** [comment]
Looks good to me, +1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1382/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Icehouse to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test failure for Icehouse to Juno migration
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-icehouse-to-juno/396/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test failure for Grizzly to Icehouse migration
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-grizzly-to-icehouse/1481/

**Comment by MirantisWorkloadMobility** [comment]
Functional test failure for Icehouse to Juno migration
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-icehouse-to-juno/397/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Icehouse to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

File: cloudferrylib/os/network/neutron.py [hunk 2]
```diff
@@ -78,17 +78,34 @@ def read_info(self, **kwargs):
...
+                for subnet in net['subnets']:
+                    # do not include the same subnets twice
+                    if subnet['id'] in [sn['id'] for sn in subnets]:
+                        continue
+
+                    LOG.debug("Append shared subnet ID %s", subnet['id'])
+                    subnets.append(self.convert_subnets(subnet, self.cloud))
+
+        routers = []
+        subnet_ids = set([sn['id'] for sn in subnets])
```
Comment (miarmak) [inline cloudferrylib/os/network/neutron.py]: We can simplify it little bit and directly use set comprehension here:
```{sn['id'] for sn in subnets}```

File: cloudferrylib/os/network/neutron.py [hunk 3]
```diff
@@ -78,17 +78,34 @@ def read_info(self, **kwargs):
         if self.filter_tenant_id is not None:
             shared_nets = self.get_shared_networks_raw()
             for net in shared_nets:
-                LOG.debug("append network ID {}".format(net['id']))
-
                 # do not include the same network twice
                 if net['id'] in [n['id'] for n in nets]:
                     continue
 
+                LOG.debug("Append shared network ID %s", net['id'])
```
Comment (miarmak) [inline cloudferrylib/os/network/neutron.py]: Maybe it is better to change `Append` to smth like `Getting` for avoid misunderstanding? Cause it is `read_info` method and we are actually do not upload/create anything here)

Comment (SVilgelm) [inline cloudferrylib/os/network/neutron.py]: This mean that we are appending the shared network into the info dictionary.

Comment (miarmak) [inline cloudferrylib/os/network/neutron.py]: Yeah, I got it. I mean that it is not obvious for those, who have not worked with CF before... I've already received some questions about such log messages...

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:1424: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1434: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1436: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1437: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1438: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1439: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1440: [W0311(bad-indentation), ] Bad indentation. Found 16 spaces, expected 12
cloudferrylib/os/network/neutron.py:1441: [W0311(bad-indentation), ] Bad indentation. Found 20 spaces, expected 16
cloudferrylib/os/network/neutron.py:166: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:276: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 21)
cloudferrylib/os/network/neutron.py:744: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:848: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:873: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:904: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:965: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1356: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1363: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1367: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1400/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Icehouse to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:1424: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1434: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1436: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1437: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1438: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1439: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1440: [W0311(bad-indentation), ] Bad indentation. Found 16 spaces, expected 12
cloudferrylib/os/network/neutron.py:1441: [W0311(bad-indentation), ] Bad indentation. Found 20 spaces, expected 16
cloudferrylib/os/network/neutron.py:166: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:276: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 21)
cloudferrylib/os/network/neutron.py:744: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:848: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:873: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:904: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:965: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1356: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1363: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1367: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1414/

**Comment by MirantisWorkloadMobility** [comment]
Functional test failure for Icehouse to Juno migration
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-icehouse-to-juno/421/

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Juno migration

**Comment by MirantisWorkloadMobility** [comment]
Functional test success for Grizzly to Icehouse migration

**Comment by roman-verchikov** [comment]
+1, can be merged.

**Pull request merged**